internal/consensus/byzantine_test.go

```diff
@@ -286,7 +288,8 @@ func TestByzantinePrevoteEquivocation(t *testing.T) {
 				assert.Equal(t, prevoteHeight, ev.Height())
 			}
 		}
-	case <-time.After(20 * time.Second):
+	case <-time.After(timeout):
+		t.Logf("Evidence from each validator: %v", evidenceFromEachValidator) // Log evidence
```

Comment:
Have you seen `evidenceFromEachValidator` containing anything other than `[<nil> <nil> <nil> <nil>]` in case of failure?

Reply:
no, this is all I've seen.

Reply:
This indicates that, after that time, no evidence was committed to a block. It is pretty unlikely (although possible) that some nodes have seen that block, while others don't. In this case the output will contain non-nil values.